src/main/utility/childpid_watcher.rs

```diff
@@ -166,51 +192,133 @@ impl ChildPidWatcher {
         }
     }
 
-    /// Call `callback` exactly once from another thread after the child `pid`
+    unsafe fn fork_watchable_internal(
+        &self,
+        fork_syscall: i64,
+        child_fn: impl FnOnce(),
+    ) -> Result<Pid, nix::Error> {
```

Comment:
It's not clear to me why `fork_watchable_internal()` or `fork_watchable()` is `unsafe`.